hostfactory/1.1/providerplugins/azurecc/scripts/src/cluster.py

```diff
@@ -13,28 +13,85 @@
 
 class Cluster:
     
+
     def __init__(self, cluster_name, provider_config, logger=None):
         self.cluster_name = cluster_name
         self.provider_config = provider_config
         self.logger = logger or logging.getLogger()
+        CC_CONFIG = {}
+        CC_CONFIG["url"] = self.provider_config.get("cyclecloud.config.web_server")
+        CC_CONFIG["username"] = self.provider_config.get("cyclecloud.config.username")
+        CC_CONFIG["password"] = self.provider_config.get("cyclecloud.config.password")
+        CC_CONFIG["cluster_name"] = self.cluster_name
+        self.node_mgr = new_node_manager(CC_CONFIG)
     
     def status(self):
         return self.get("/clusters/%s/status" % self.cluster_name)
+    
+    #old add nodes function
+    # def add_nodes(self, request):
+    #     self.logger.debug(request)
+    #     response_raw = self.post("/clusters/%s/nodes/create" % self.cluster_name, json=request)
+    #     try:
+    #         return json.loads(response_raw)
+    #     except:
+    #         raise RuntimeError("Could not parse response as json to create_nodes! '%s'" % response_raw)
+    
+    def add_nodes(self,request):
+        sku=request['sets'][0]['definition']['machineType']
+        selector = {"node.nodearray": request['sets'][0]['nodearray']}
+        if sku:
+          selector["node.vm_size"] = sku
+        self.logger.debug(request)
+        r=self.node_mgr.allocate(selector, node_count=request['sets'][0]['count'])
+        self.logger.debug("Request id in add nodes %s",request['requestId'])
 
-    def add_nodes(self, request):
-        response_raw = self.post("/clusters/%s/nodes/create" % self.cluster_name, json=request)
-        try:
-            return json.loads(response_raw)
-        except:
-            raise RuntimeError("Could not parse response as json to create_nodes! '%s'" % response_raw)
+        request_id_start = f"{request['requestId']}-start"
+        request_id_create = f"{request['requestId']}-create"
+        bootpup_resp=self.node_mgr.bootup(
```

Comment:
be careful with formatting. 'var = expr', not 'var=expr'

Reply:
ok yeah will update it